### PR TITLE
mod: Upgrade to ginkgo v2

### PR DESCRIPTION
**Pull request by haoxins**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,15 +10,16 @@ require (
...
 	github.com/moverest/mnist v0.0.0-20160628192128-ec5d9d203b59
-	github.com/onsi/ginkgo v1.16.4
-	github.com/onsi/gomega v1.15.0
+	github.com/onsi/ginkgo/v2 v2.1.3
+	github.com/onsi/gomega v1.18.1
 	github.com/operator-framework/operator-lib v0.8.0
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.52.0
 	github.com/spf13/viper v1.7.1
 	github.com/stretchr/testify v1.7.0
 	github.com/tommy351/goldga v0.3.0
```
Comment (pvaneck) [inline go.mod:19]: This should be updated to v0.5.0 for ginkgo v2 support. I believe that's why the unit tests are failing.

**Review by pvaneck — COMMENTED** [review]
Thanks for the upgrading ginkgo! I left an inline comment that should get the unit tests passing.

File: go.mod (hunk 1 above)
Comment (haoxins) [reply go.mod]: Thanks, and tests passed.

**Comment by pvaneck** [comment]
Thanks, @haoxins!
/lgtm
/approve

**Pull request merged**